### PR TITLE
refactor(admin): visually update edit form item dialog

### PR DESCRIPTION
**Pull request by xflord**
* updated the dialog for editing application form items to match the vertical layout of other dialogs

### CODE REVIEW

File: apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.html [hunk 1]
```diff
@@ -10,163 +10,155 @@ <h1 mat-dialog-title>
...
+                (itemSelected)="this.disabledDependencyItem=$event"
+                [label]="'DIALOGS.APPLICATION_FORM_EDIT_ITEM.DISABLED_DEPENDENCY' | translate"
+                [hint]="'DIALOGS.APPLICATION_FORM_EDIT_ITEM.DISABLED_DEPENDENCY_DESCRIPTION' | translate"
                 [items]="possibleDependencyItems">
               </perun-web-apps-app-form-item-search-select>
             </div>
-          </app-edit-application-form-item-line>
+          </div>
+          <!-- Hidden part -->
+          <mat-form-field class="w-100">
```
Comment (HejdaJakub) [inline apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.html:141]:
```suggestion
          <mat-form-field subscriptSizing="dynamic" class="w-100">
```

File: apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.html [hunk 2]
```diff
@@ -175,62 +167,63 @@ <h1 mat-dialog-title>
...
-              </mat-form-field>
-            </app-edit-application-form-item-line>
-            <app-edit-application-form-item-line
-              [description]="'DIALOGS.APPLICATION_FORM_EDIT_ITEM.ERROR_DESCRIPTION' | translate"
-              [label]="'DIALOGS.APPLICATION_FORM_EDIT_ITEM.ERROR' | translate">
-              <mat-form-field class="w-100">
-                <textarea formControlName="{{lang}}-plain-error-message" matInput></textarea>
-              </mat-form-field>
-            </app-edit-application-form-item-line>
+            <mat-form-field class="w-100">
```
Comment (HejdaJakub) [inline apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.html:188]:
```suggestion
            <mat-form-field subscriptSizing="dynamic" class="w-100">
```

File: apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.html [hunk 3]
```diff
@@ -175,62 +167,63 @@ <h1 mat-dialog-title>
...
+              <mat-label>{{'DIALOGS.APPLICATION_FORM_EDIT_ITEM.LABEL' | translate}}</mat-label>
+              <textarea formControlName="{{lang}}-plain-label" matInput></textarea>
+              <mat-hint
+                >{{'DIALOGS.APPLICATION_FORM_EDIT_ITEM.LABEL_DESCRIPTION' | translate}}</mat-hint
+              >
+            </mat-form-field>
+            <mat-form-field class="w-100">
```
Comment (HejdaJakub) [inline apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.html:195]:
```suggestion
            <mat-form-field subscriptSizing="dynamic" class="w-100">
```

**Review by HejdaJakub — CHANGES_REQUESTED** [review]
I suggested some places where `subscriptSizing="dynamic"` also makes sense (at least for mobile screens).

I am also not sure if the warning displayed for attribute relation looks the best now - it seems to be huge in comparison with the rest of items in the dialog (and also with the previous view).
_Now:_ image
_Previously:_ image

**Comment by xflord** [comment]
I added `subscriptSizing="dynamic"` to form field with hints.
When it comes to the warning - I first tried to add a mat-error element, however with the way EntitySearchSelectComponent is written it did not seem to work when I tried (and I tried...). Considering that this action could break the form, I assumed the size wouldn't be that much of a problem (and also would be consistent with other warnings?). But changing it to a red paragraph shouldn't be a problem. What do you think @HejdaJakub @bodnara

File: apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.html [hunk 4]
```diff
@@ -10,163 +10,169 @@ <h1 mat-dialog-title>
...
-            [label]="'DIALOGS.APPLICATION_FORM_EDIT_ITEM.INPUT_WIDGET' | translate">
-            {{applicationFormItem.type | applicationFormItemType}}
-          </app-edit-application-form-item-line>
-          <app-edit-application-form-item-line
-            [description]="'DIALOGS.APPLICATION_FORM_EDIT_ITEM.DISPLAY_ON_APPLICATION_DESCRIPTION' | translate"
-            [label]="'DIALOGS.APPLICATION_FORM_EDIT_ITEM.DISPLAY_ON_APPLICATION' | translate">
-            <section>
+          <mat-form-field subscriptSizing="dynamic" class="w-100">
+            <mat-label>{{'DIALOGS.APPLICATION_FORM_EDIT_ITEM.SHORT_NAME' | translate}}</mat-label>
+            <input [(ngModel)]="applicationFormItem.shortname" matInput readonly />
```
Comment (HejdaJakub) [inline apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.html:15]: Shortname should be editable.
```suggestion
            <input [(ngModel)]="applicationFormItem.shortname" matInput />
```

**Review by HejdaJakub — CHANGES_REQUESTED** [review]
The sections made it much better, I really like it! :raised_hands:

File: apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.html (hunk 4 above)
Comment (xflord) [reply apps/admin-gui/src/app/shared/components/dialogs/edit-application-form-item-dialog/edit-application-form-item-dialog.component.html:15]: Was testing how it looks for the item type and forgot to change it back, nice catch

**Review by bodnara — APPROVED** [review]
Nicely done, I would say quite an improvement.

**Review by HejdaJakub — APPROVED** [review]

**Pull request merged**